net/nimble/host/src/ble_att.c

```diff
@@ -452,7 +452,7 @@ ble_att_chan_mtu(const struct ble_l2cap_chan *chan)
 
         mtu = BLE_ATT_MTU_DFLT;
     } else {
-        mtu = min(chan->my_mtu, chan->peer_mtu);
+        mtu = min(ble_att_preferred_mtu_val, chan->peer_mtu);
```

Comment:
Hi Chris,

Ahh I missed that preferred MTU can be changed by user any time and that it actually set chan->my_mtu which is not what I wanted.
My goal is to make sure chan->my_mtu is actuall MTU as negotiated over the air.
Main reason for this is to make sure channel does not get more data on the channel than it is allowed.

Anyway, I need to keep track on MTU sent over the air to remote device. I will use proc from gattc as exchange procedure is attached to that.  BTW I would move it to ble_att but this is topic for later maybe.
When it comes to  unit tests, I did some helper function for exchange mtu procedure as now we cannot just set chan->my_mtu without exchanging with peer.
I think that should work like this.

Thanks
Łukasz